test/features/app/app.js

```diff
@@ -2,21 +2,23 @@
 
 'use strict';
 
-
 const preq   = require('preq');
 const assert = require('../../utils/assert.js');
-const server = require('../../utils/server.js');
-
-if (!server.stopHookAdded) {
-    server.stopHookAdded = true;
-    after(() => server.stop());
-}
+const Server = require('../../utils/server.js');
 
-describe('express app', function() {
+describe('express app', function () {
 
     this.timeout(20000); // eslint-disable-line no-invalid-this
 
-    before(() => server.start());
+    const server = new Server();
+
+    before(() => {
+        return server.start();
+    });
+
+    after(() => {
+        server.stop();
+    });
```

Comment:
1. Nit: `before(() => server.start());` one-liner seems nicer too me. 

2. The `server.stop()` returns a promise so it needs to be returned so that we wait for it to be resolved before starting the next text suite. One-line syntax would help here too